### ISSUE TITLE
Legacy HistoryFloat(..) error when training

### ISSUE THREAD

**Issue by firefox7025** [deeplearning4j/deeplearning4j#6492]
Hi there, I seem to be getting some weird issues when training training. 
It said error in Legacy HistoryFloat(..) mid training set full stack trace attached.

Network is Vgg16 non-pretrained with 53 total labels. 

[log.txt](https://github.com/deeplearning4j/deeplearning4j/files/2419647/log.txt)
 Any ideas?

**Comment by AlexDBlack** [comment]
OK, so where too begin here...

Right from the start you are running into memory issues. You'll note these throughout your entire training process:
> 18:47:39.905 [main] WARN org.nd4j.jita.handler.impl.CudaZeroHandler - Out of [DEVICE] memory, host memory will be used instead: deviceId: [0], requested bytes: [2360320]

My guess is this is the underlying cause of the later op failure.
Given the network architecture and hardware (1080 GPU) I would not have expected you to be running into memory issues.

Can you post your training code, and post the DL4J version you are using?

**Comment by firefox7025** [comment]
I did create another branch without altering the workspaces and that does seem to have removed all the memory warnings. I am rerunning it now, so in about 12 hours or so if its still in the green I will take your initial gut feeling as correct.

**Comment by AlexDBlack** [comment]
I had a quick look through your code - no obvious issues. You have workspaces (though use Workspace.ENABLED instead of SINGLE - single is deprecated). And you're on 1.0.0-beta2.

If you are still having trouble, maybe post the output of:
https://github.com/deeplearning4j/deeplearning4j/blob/27360a7e29681bb90ee994892cd302614bb9b952/deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/multilayer/MultiLayerNetwork.java#L3548-L3562

And one other idea here is other (non-DL4J) programs using GPU memory. That should show up in nvidia-smi.

**Comment by firefox7025** [comment]
There were no other programs using it, however after a few tweaks and playing with gc collection times, it seems to be working fine now. No memory warnings and no legacy history failures after 12 hours of training time. Calling this issue closed, thank you for the help!
